boards/common/stm32/include/l4/cfg_clock_default.h

```diff
@@ -0,0 +1,187 @@
+     IS_ACTIVE(CONFIG_USE_CLOCK_HSE))
+#error "Cannot use HSI as clock source with other clock configurations"
+#endif
+
+#ifndef CONFIG_BOARD_HAS_HSE
+#define CONFIG_BOARD_HAS_HSE            0
+#endif
+#ifndef CLOCK_HSE
+#define CLOCK_HSE                       MHZ(8)
+#endif
+#if IS_ACTIVE(CONFIG_BOARD_HAS_HSE) && (CLOCK_HSE < MHZ(4) || CLOCK_HSE > MHZ(48))
+#error "HSE clock frequency must be between 4MHz and 48MHz"
+#endif
+
+#ifndef CONFIG_BOARD_HAS_LSE
+#define CONFIG_BOARD_HAS_LSE            0
+#endif
+#if IS_ACTIVE(CONFIG_BOARD_HAS_LSE)
+#define CLOCK_LSE                       (1)
+#else
+#define CLOCK_LSE                       (0)
+#endif
+
+#define CLOCK_HSI                       MHZ(16)
+
+#ifndef CONFIG_CLOCK_MSI
+#define CONFIG_CLOCK_MSI                MHZ(48)
+#endif
+
+/* The following parameters configure a 80MHz system clock with PLL as input clock */
+#ifndef CONFIG_CLOCK_PLL_SRC_MSI
+#if IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_HSE) || IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_HSI) || \
+    CONFIG_BOARD_HAS_HSE
+#define CONFIG_CLOCK_PLL_SRC_MSI        0
+#else
+#define CONFIG_CLOCK_PLL_SRC_MSI        1       /* Use MSI an input clock by default */
+#endif
+#endif /* CONFIG_CLOCK_PLL_SRC_MSI */
+#ifndef CONFIG_CLOCK_PLL_SRC_HSE
+#if IS_ACTIVE(CONFIG_BOARD_HAS_HSE)
+#define CONFIG_CLOCK_PLL_SRC_HSE        1
+#else
+#define CONFIG_CLOCK_PLL_SRC_HSE        0
+#endif
+#endif
+#ifndef CONFIG_CLOCK_PLL_SRC_HSI
+#define CONFIG_CLOCK_PLL_SRC_HSI        0
+#endif
+#ifndef CONFIG_CLOCK_PLL_M
+#define CONFIG_CLOCK_PLL_M              (6)
+#endif
+#ifndef CONFIG_CLOCK_PLL_N
+#define CONFIG_CLOCK_PLL_N              (20)
+#endif
+#ifndef CONFIG_CLOCK_PLL_R
+#define CONFIG_CLOCK_PLL_R              (2)
+#endif
+
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSI)
+#define CLOCK_CORECLOCK                 (CLOCK_HSI)
+
+#elif IS_ACTIVE(CONFIG_USE_CLOCK_HSE)
+#if !IS_ACTIVE(CONFIG_BOARD_HAS_HSE)
+#error "The board doesn't provide an HSE oscillator"
+#endif
+#define CLOCK_CORECLOCK                 (CLOCK_HSE)
+
+#elif IS_ACTIVE(CONFIG_USE_CLOCK_MSI)
+#define CLOCK_CORECLOCK                 (CONFIG_CLOCK_MSI)
+
+#elif IS_ACTIVE(CONFIG_USE_CLOCK_PLL)
+#if IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_MSI)
+#define CLOCK_PLL_SRC                   (CONFIG_CLOCK_MSI)
+#elif IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_HSE)
+#define CLOCK_PLL_SRC                   (CLOCK_HSE)
+#else /* CONFIG_CLOCK_PLL_SRC_ */
+#define CLOCK_PLL_SRC                   (CLOCK_HSI)
+#endif
+#define CLOCK_CORECLOCK                 \
+        ((CLOCK_PLL_SRC / CONFIG_CLOCK_PLL_M) * CONFIG_CLOCK_PLL_N) / CONFIG_CLOCK_PLL_R
+#ifndef CLOCK_CORECLOCK_MAX
+#define CLOCK_CORECLOCK_MAX             MHZ(80)
+#endif
+#if CLOCK_CORECLOCK > CLOCK_CORECLOCK_MAX
+#if CLOCK_CORECLOCK_MAX == MHZ(64)
+#error "SYSCLK cannot exceed 64MHz"
+#elif CLOCK_CORECLOCK_MAX == MHZ(80)
+#error "SYSCLK cannot exceed 80MHz"
+#elif CLOCK_CORECLOCK_MAX == MHZ(120)
+#error "SYSCLK cannot exceed 120MHz"
```

Comment:
Why not use the `XTSTR` for this?

Reply:
It cannot be used in the preprocessor code itself.

Reply:
I think it can be doenwith `pragma`, but Ok.